### PR TITLE
Allow user to comment on individual answer

### PR DESCRIPTION
**Pull request by kxmbrian**
Previously, a user would have to save the whole submission in order to comment on an individual answer.

discuss each answer

### CODE REVIEW

**Comment by weiqingtoh** [comment]
I think this will have merge conflicts with #1164 >_< 

To @allenwq and @kxmbrian, just be careful when we merge (either 1 or the other), then the remaining PR will have to be rebased.

**Comment by weiqingtoh** [comment]
Oh just ignore what I said >_< I realised they don't clash - that is a good thing 👍

File: app/controllers/course/assessment/submission/answer/comments_controller.rb [hunk 1]
```diff
@@ -7,7 +7,23 @@ class Course::Assessment::Submission::Answer::CommentsController < \
 
   delegate :discussion_topic, to: :@answer
 
+  def create
+    @answer.class.transaction do
+      @post.title = @assessment.title
+      if super && @answer.save
```
Comment (weiqingtoh) [inline app/controllers/course/assessment/submission/answer/comments_controller.rb]: Do you mean to use `create_topic_subscription` here? Or am I missing something?

**Comment by weiqingtoh** [comment]
Other than the location of the `Form_Helper` js file, LGTM 👍

File: app/controllers/course/assessment/submission/answer/comments_controller.rb (hunk 1 above)
Comment (kxmbrian) [inline app/controllers/course/assessment/submission/answer/comments_controller.rb]: This is called in `Course::Discussion::PostsConcern#create`. When a user comments on an answer, he will be automatically subscribed to it.

Comment (weiqingtoh) [inline app/controllers/course/assessment/submission/answer/comments_controller.rb]: Got it - I realised I went to a different `PostsConcern`, where I didn't see the `create` function.

File: app/assets/javascripts/helpers/form_helpers.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    });
+
+    return data;
+  };
+
+
+  /**
+   * @callback formSubmitSuccessCallback
+   * @param {data} data The server response
+   * @param {HTMLElement} form The submitted form
```
Comment (allenwq) [inline app/assets/javascripts/helpers/form_helpers.js]: What's this for ?  (From line 47 to 57)

Comment (kxmbrian) [inline app/assets/javascripts/helpers/form_helpers.js]: They are referenced in line 59-65. What is the right way to document `submitAndDisableForm`?

Comment (allenwq) [inline app/assets/javascripts/helpers/form_helpers.js]: OK then, seems you are correct from JS doc. But document this way a bit overkill to me ><

**Comment by allenwq** [comment]
@kxmbrian Should you hide the button or let the button do nothing when JS is disabled ?

**Comment by kxmbrian** [comment]
The latest update hides the form

**Comment by allenwq** [comment]
Thanks ! looks good ! @weiqingtoh any suggestions for the location of js helper file ?

**Comment by kxmbrian** [comment]
I've shifted the helper to be in the right place already

**Comment by weiqingtoh** [comment]
Location seems fine - can merge? @allenwq

**Comment by allenwq** [comment]
Yep !

**Pull request merged**